wagtail/wagtailcore/models.py

```diff
@@ -242,6 +242,11 @@ def specific(self):
         return self.get_queryset().specific()
 
 
+class ProxyPageManager(PageManager):
+    def get_queryset(self):
+        return super().get_queryset().type(self.model)
```

Comment:
``super().get_queryset()`` should return pages with the proxy model class and ``type(self.model)`` should filter them to only include items that are members of that proxy model.

But you're right, I missed ``for_concrete_models=False`` in the ``.type()`` implementation causing that type filter to select pages with the parent type (but still return instances of the proxy model).. Doh!